### PR TITLE
#143. Logfile name includes uname per run

### LINKED ISSUE #143: set the log file name to include uname

### PR DESCRIPTION
**Pull request by Raa37**
Testing methods:
- ran multiple runs, each run produces a log file -> uname_time.log of that run and everything is logged there.

We have an Atlas.log that contains log messages of anything from runSim.py, aka. anything that happens before individual runs are kicked off.

### CODE REVIEW

File: LoggingConfig.py [hunk 1]
```diff
@@ -22,7 +22,7 @@
             'level':              'DEBUG',
             'formatter':          'formatter_file',
             'class':              'logging.handlers.RotatingFileHandler',
-            'filename':           'Atlas.log',
+            'filename':           './logs/Atlas.log',
```
Comment (twatteyne) [inline LoggingConfig.py]: revert this change

File: Utils.py [hunk 2]
```diff
@@ -84,3 +89,12 @@ def get(self):
         self.check.remove(item)
         return item
 
+# ============== Logging Setup
+
+
+def SetLoggerUname(uname):
```
Comment (twatteyne) [inline Utils.py]: rename to `setLoggerUname`, by convention functions do not start with a capital letter

File: Utils.py [hunk 3]
```diff
@@ -84,3 +89,12 @@ def get(self):
...
+    LoggingConfig.LOGGINGCONFIG['handlers']['handler_file']['filename'] = './logs/{}_{}.log'.format(uname, time.strftime("%y%m%d%H%M%S", time.localtime()))
+    logging.config.dictConfig(LoggingConfig.LOGGINGCONFIG)
+    return logging.getLogger('RunOneSim')
```
Comment (twatteyne) [inline Utils.py]: abstraction violation. Your function is called `SetLoggerUname()`, it can NOT secretely create a logger which happens to be called `RunOneSim`. remove the return statement, call `logging.getLogger('RunOneSim')` from `RunOneSim`

**Review by twatteyne — CHANGES_REQUESTED** [review]
Is this code working dependent on the order modules are initialized and when `setLoggerUname()` is called?

File: Utils.py (hunk 3 above)
Comment (Raa37) [reply Utils.py]: I did it in a slightly different way, I just wanted to ask if it's valid, otherwise will change to what you asked for. I just thought this way we have less log related lines in RunOneSim and we also don't have an abstraction violation in Utils.

File: Utils.py [hunk 4]
```diff
@@ -84,3 +89,12 @@ def get(self):
         self.check.remove(item)
         return item
 
+# ============== Logging Setup
+
```
Comment (twatteyne) [inline Utils.py:93]: remove newline

File: Utils.py [hunk 5]
```diff
@@ -84,3 +89,12 @@ def get(self):
...
+
+def setLoggerUname(uname, loggername=None):
+    LoggingConfig.LOGGINGCONFIG['handlers']['handler_file']['filename'] = './logs/{}_{}.log'.format(uname, time.strftime("%y%m%d%H%M%S", time.localtime()))
+    logging.config.dictConfig(LoggingConfig.LOGGINGCONFIG)
+    return logging.getLogger(loggername)
```
Comment (twatteyne) [inline Utils.py]: no, please implement what I suggested

**Review by twatteyne — CHANGES_REQUESTED** [review]

**Review by twatteyne — APPROVED** [review]
